### ISSUE TITLE
New rule idea: Similar to S3262 (MethodOverrideNoParams) but for interfaces

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6444]
Similar to  S3262, params should not be hidden for **interfaces**, as shown here:

```csharp
Interface @interface = new Class();
Class actual = new Class();

@interface.Method(1,2,42);
actual.Method(new int[] {1,2,42}); 

public interface Interface
{
    public void Method(params int[] things);
}

public class Class : Interface
{
    public void Method(int[] things) // This should raise, hides params
    {
        throw new NotImplementedException();
    }
}
```

Furthermore, the compiler does not generate the necessary parts for `params` invocations to work on the class, like it does on class inheritance. For example:

```csharp
@interface.Method(); // working as expected, can be called empty.
actual.Method(); // Build error, cannot be called empty, [params] is not implicit.
actual.Method(1,2,42); // Build error, cannot be called with (arg1,arg2,arg3,..).

```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1799](https://sonarsource.atlassian.net/browse/NET-1799)

[NET-1799]: https://sonarsource.atlassian.net/browse/NET-1799?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
